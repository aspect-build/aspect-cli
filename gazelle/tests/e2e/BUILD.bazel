load("@bazel_gazelle//:def.bzl", "gazelle_binary")
load("//gazelle:gazelle.bzl", "gazelle_generation_test")

# Disable the gazelle extensions
# gazelle:exclude .

# Starzelle e2e tests with all standard gazelle extensions enabled.
# Each test has its own additional *.star plugins.

gazelle_binary(
    name = "gazelle_binary-host_js",
    languages = [
        "//gazelle/js",
        "//gazelle/host",
    ],
    visibility = ["//visibility:private"],
)

[
    gazelle_generation_test(
        name = "%s_test" % t,
        dir = t,
        env = {
            "TEST_STARZELLE_PLUGINS": "%s/%s" % (
                package_name(),
                t,
            ),
        },
        gazelle_binary = ":gazelle_binary-host_js",
    )
    for t in [t.replace("/WORKSPACE", "") for t in glob(["*/WORKSPACE"])]
]
