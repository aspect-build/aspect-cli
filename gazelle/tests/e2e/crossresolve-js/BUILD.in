
# aspect:js_files viteconfig vitest.config.mjs
