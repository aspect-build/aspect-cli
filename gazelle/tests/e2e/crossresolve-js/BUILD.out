load("@aspect_rules_js//js:defs.bzl", "js_library")
load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
load("@deps-test//my:rules.bzl", "x_lib")

# aspect:js_files viteconfig vitest.config.mjs

ts_project(
    name = "crossresolve-js",
    srcs = ["main.ts"],
    deps = [
        ":a",
        ":b",
        "//sub:a",
    ],
)

x_lib(name = "a")

js_library(
    name = "b",
    srcs = ["file-served-by-js_library.js"],
)
